sensors: voltage_divider: Reconfigure ADC for the channel to be used

guenzel-kinexon opened the pull request:
Some ADCs (e.g. SAM0) can only convert one channel at a time and therefore need to be reconfigured for every voltage divider prior to startin the conversion.

In the current implementation only one channel (whichever was initialized last) will be converted.

Fixes: #76732

MaureenHelm reviewed (changes requested):
Please squash the fixup commit into the original commit

guenzel-kinexon committed:
sensors: reconfig adc for channel

    Some ADCs (e.g. SAM0) can only convert one channel at a time
    and therefore need to be reconfigured for every voltage
    divider prior to startin the conversion

Signed-off-by: Patrick Günzel <patrick.guenzel@kinexon.com>

guenzel-kinexon commented:
> Please squash the fixup commit into the original commit

done

ubieda commented on drivers/sensor/voltage_divider/voltage.c:
```diff
@@ -171,7 +179,7 @@ static int voltage_init(const struct device *dev)
 	PM_DEVICE_DT_INST_DEFINE(inst, pm_action);                                                 \
                                                                                                    \
 	SENSOR_DEVICE_DT_INST_DEFINE(inst, &voltage_init, PM_DEVICE_DT_INST_GET(inst),             \
-			      &voltage_##inst##_data, &voltage_##inst##_config, POST_KERNEL,       \
-			      CONFIG_SENSOR_INIT_PRIORITY, &voltage_api);
+				     &voltage_##inst##_data, &voltage_##inst##_config,             \
+				     POST_KERNEL, CONFIG_SENSOR_INIT_PRIORITY, &voltage_api);
```
Non-related change, it seems.

guenzel-kinexon replied:
was done by the auto-formatter. Reverted the change

ubieda commented on drivers/sensor/voltage_divider/voltage.c:
```diff
@@ -74,8 +82,8 @@ static int get(const struct device *dev, enum sensor_channel chan, struct sensor
 	/* Note if full_ohms is not specified then unscaled voltage is returned */
 	(void)voltage_divider_scale_dt(&config->voltage, &v_mv);
 
-	LOG_DBG("%d of %d, %dmV, voltage:%dmV", data->raw,
-		(1 << data->sequence.resolution) - 1, raw_val, v_mv);
+	LOG_DBG("%d of %d, %dmV, voltage:%dmV", data->raw, (1 << data->sequence.resolution) - 1,
+		raw_val, v_mv);
```
Non-related change, it seems.

guenzel-kinexon replied:
was done by the auto-formatter. Reverted the change

ubieda reviewed (dismissed):
LGTM but please discard non-related changes.

mmahadevan108 commented:
@guenzel-kinexon Could you please rebase and help fix the merge conflict

kartben commented:
@guenzel-kinexon friendly ping :)

guenzel-kinexon commented:
friendly pong :)

Changes should be ok now.

kartben commented:
@MaureenHelm please revisit

MaureenHelm reviewed (approved):
Not sure why the scancode check is failing.

@kartben I've been out for a couple of weeks, is this something you've seen recently that needs a rebase to resolve?

kartben commented:
ya I was wondering the same -- CODEOWNERS recently got nuked but I'm not sure why it's even relevant since it's not touched by this PR. Just rebased to trigger the failing CI job since it was too old to be re-triggered manually.
